client/post-editor/media-modal/detail/detail-item.jsx

```diff
@@ -59,6 +53,33 @@ const enableImageEditing = ( item, site ) => {
 	return true;
 };
 
+/**
+ * This function return true if the video editor can be
+ * enabled/shown
+ *
+ * @param  {object} item - media item
+ * @param  {object} site - current site
+ * @return {boolean} `true` if the video editor can be enabled.
+ */
+const enableVideoEditing = ( item, site ) => {
+	// do not show if the feature flag isn't set
+	if ( ! config.isEnabled( 'post-editor/video-editor' ) ) {
+		return false;
+	}
+
+	// do not allow for Jetpack site with VideoPress disabled
+	if ( get( site, 'jetpack', false ) ) {
+		if ( ! MediaUtils.isVideoPressItem( item ) ) {
+			return false;
+		}
+	// do not allow for wpcom site with VideoPress disabled
+	} else if ( ! site.options.videopress_enabled ) {
+		return false;
+	}
+
+	return true;
+};
```

Comment:
just for fun I'd like to point out that we have another way of structuring this function. it's really just one big conditional

```js
return (
	config.isEnabled( 'post-editor/video-editor' ) &&
	(
		// non-jetpack site
		( 
			! get( site, 'jetpack' ) && 
			get( site, 'options.videopress_enabled' ) 
		) ||
		// jetpack site
		MediaUtils.isVideoPressItem( item )
	)
)
```